### PR TITLE
feat: add ZoomInfo Toolkit for CAMEL-AI framework

### LINKED ISSUE #3316: [Feature Request] Integrate zoominfo as toolkit
### Required prerequisites

- [x] I have searched the [Issue Tracker](https://github.com/camel-ai/camel/issues) and [Discussions](https://github.com/camel-ai/camel/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [ ] Consider asking first in a [Discussion](https://github.com/camel-ai/camel/discussions/new).

### Motivation

integrate zoominfo as toolkit https://www.zoominfo.com/solutions/data-as-a-service/enterprise-api

### Solution

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by haoye2**
Fixes #3316

## Description
Add comprehensive ZoomInfo API integration to CAMEL-AI framework, enabling AI agents to perform B2B data intelligence, company research, and contact discovery.

## 🔧 Features
- **Company Search**: Advanced company lookup and enrichment
- **Contact Discovery**: Find contacts within target companies  
- **Authentication**: JWT and PKI support with auto-refresh
- **Error Handling**: Comprehensive error management and logging
- **Testing**: 100% test coverage with mock-based testing

## 🧪 Testing
- ✅ 9/9 tests passing (100%)
- ✅ Mock-based API testing
- ✅ Authentication flow testing
- ✅ Error condition handling
- ✅ Edge case coverage

## 📚 Documentation
- Complete README with installation and usage examples
- API documentation with type hints
- Configuration guide for environment variables
- Contributing guidelines

## ✅ Compliance
- Follows CAMEL naming conventions (`zoominfo_*` prefix)
- Uses CAMEL logging standards
- Apache 2.0 licensed
- Pre-commit hooks configured
- Modern Python project structure

## Checklist
- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have linked this PR to an issue using the Development section on the right sidebar or by adding `Fixes #3316` in the PR description (**required**)
- [x] I have checked if any dependencies need to be added or updated in `pyproject.toml` and `uv lock`
- [x] I have updated tests accordingly (*required for a bug fix or a new feature*)
- [x] I have updated documentation if needed
- [x] I have added examples if this is a new feature

### CODE REVIEW

**Comment by Wendong-Fan** [comment]
thanks @haoye2  for the PR!

File: Zoomtoolkit/camel/__init__.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# ========= Copyright 2023-2024 @ CAMEL-AI.org. All Rights Reserved. =========
+# Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (waleedalzarooni) [inline Zoomtoolkit/camel/__init__.py]: May be an error with the PR submission but I think the licensing text should already be there
